src/BaseClient.php

```diff
@@ -607,7 +608,7 @@ private function retryDecider()
             $retries,
             RequestInterface $request,
             ?ResponseInterface $response = null,
-            ?RequestException $exception = null
+            RequestException|ClientException|null $exception = null
```

Comment:
I think its better to use `\Psr\Http\Client\ClientExceptionInterface`  thats a overall interface for all the Http Exceptions

Reply:
When will this change be merged?